packages/members-api/src/controllers/router/index.js

```diff
@@ -96,23 +107,28 @@ module.exports = class RouterController {
         const subscriptionUpdateData = {
             id: subscriptionId
         };
-        if (cancelAtPeriodEnd !== undefined) {
-            subscriptionUpdateData.cancel_at_period_end = cancelAtPeriodEnd;
-            subscriptionUpdateData.cancellation_reason = cancellationReason;
-        }
 
         if (planName !== undefined) {
-            const plan = this._stripePlanService.getPlans().find(plan => plan.nickname === planName);
+            const plan = this._stripePlansService.getPlans().find(plan => plan.nickname === planName);
             if (!plan) {
                 throw new errors.BadRequestError({
                     message: 'Updating subscription failed! Could not find plan'
                 });
             }
             subscriptionUpdateData.plan = plan.id;
+            this._stripeAPIService.changeSubscriptionPlan(subscriptionId, plan.id);
+        } else if (cancelAtPeriodEnd !== undefined) {
```

Comment:
I think this is wrong? I thought that we need to be able to both update a plan AND uncancel the subscription for portal functionality?

Reply:
We should be `await`ing any async calls too

Reply:
🤔 Isn't that the implicit handling of `changeSubscriptionPlan` method where its automatically setting `cancel_at_period_end` to `false` ? So if a plan is being changed, cancel at period end is always set to `false` and otherwise its handled as is ?

Reply:
Aha yeah - good point! We still should await any async method tho!